File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.js
function solution(code) {
    let mode = 0;
    let ret = '';
    
    for(let i=0; i<code.length; i++){
        if(mode === 0){
            if(code[i] !== "1"){
                if(i % 2 === 0){
                    ret += code[i];
                }
            }else{
                mode = 1;
            }
        }else{
            if(code[i] !== "1"){
                if(i % 2 === 1){
                    ret += code[i];
                }
            }else{
                mode = 0;
            }
        }
        
        
    }
  
        
    if(ret.length == 0){
        return 'EMPTY';
    }
    return ret;
}

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    answer = ''
    
    mode = 0
    
    for idx,str in enumerate(code):
        if mode == 0:
            if code[idx] != '1' and idx % 2 == 0:
                answer+=(code[idx])
            if code[idx] == '1' :
                mode = 1
        else:
            if code[idx] != '1' and idx % 2 == 1:
                answer+=(code[idx])
            if code[idx] == '1':
                mode = 0
    return 'EMPTY' if len(answer) == 0 else answer